src/test/java/emailvalidator4j/lexer/EmailLexerTest.java

```diff
@@ -167,6 +167,17 @@ public void toStringReturnsLeftTokens() {
         Assert.assertTrue("@bar.com".equals(lexer.toString()));
     }
 
+    @Test
+    @UseDataProvider("atextExamples")
+    public void atextParsing(String atext) {
+        EmailLexer lexer = new EmailLexer();
+        lexer.lex(atext);
+        TokenInterface token = lexer.getCurrent();
+        Assert.assertEquals(Tokens.GENERIC, token.getName());
+        Assert.assertEquals(atext, token.getText());
```

Comment:
Insist that atext be parsed as a single token even with mixed kinds of characters in it. Avoid parser regressions.